### PR TITLE
feat: use cert-manager for SSL certificates

### PR DESCRIPTION
**Pull request by viveksyngh**
fixes: #213
Signed-off-by: Vivek Singh <svivekkumar@vmware.com>

### CODE REVIEW

**Review by prometherion — CHANGES_REQUESTED** [review]
Please, bump up the chart in order to get it published.

We have this condition and we have to avoid overwriting the previous version.

https://github.com/clastix/capsule-proxy/blob/7e711747a93d664336a562aa25c5f84f1362be39/.github/workflows/helm.yaml#L20

**Comment by viveksyngh** [comment]
Updated the chart version to `0.2.5`

File: charts/capsule-proxy/templates/certmanager.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    kind: Issuer
+    group: cert-manager.io
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: {{ include "capsule-proxy.fullname" . }}-ca-issuer
+spec:
+  ca:
+    secretName: {{ include "capsule-proxy.fullname" . }}-root-secret
```
Comment (carpenterm) [inline charts/capsule-proxy/templates/certmanager.yaml:31]: There are use cases where we may want to use an existing CA - say for example, the Kubernetes CA. Could we add some flag to enable/disable the self-signed CA certs and allow the user to provide the secret name used in the CA Issuer via the chart values i.e. the one at `spec.ca.secretName` above?

E.g.

```
certManager:
  generateCertificates: true
  existingCA:
    enabled: true
    secretName: my-custom-ca-cert
```

Comment (carpenterm) [reply charts/capsule-proxy/templates/certmanager.yaml:31]: Thinking about it, `externalCA` may be a better name than `existingCA`. Just to add, the `certManager.externalCA.secretName` (or whatever we call it) would need to point to a secret containing the CA cert and private key of the external CA in the correct cert-manager format as per https://cert-manager.io/docs/configuration/ca/#deployment

Comment (prometherion) [reply charts/capsule-proxy/templates/certmanager.yaml:31]: A helper could be implemented, something as the following:

```
{{- define "capsule-proxy.caSecretName" }}
{{- if .Values.ca.secretName }}
{{- printf "%s" .Values.ca.secretName }}
{{- else }}
{{- printf "%s-root-secret" (include "capsule-proxy.fullname" .) }}
{{- end }}
```

**Pull request merged**